tealish/nodes.py

```diff
@@ -24,12 +32,18 @@ def __init__(self, line, parent=None, compiler=None) -> None:
         self._line = line
         self._line_no = compiler.line_no if compiler else None
         # self.child_nodes includes nested nodes (e.g. function body or statements within if...else...end)
-        self.child_nodes = []
+        self.child_nodes: List[BaseNode] = []
         # self.nodes includes structural nodes and child_nodes (e.g. function args and body, if conditions and child statements)
-        self.nodes = []
+        self.nodes: List[BaseNode] = []
         self.properties = {}
         try:
-            self.matches = re.match(self.pattern, self.line).groupdict()
+            matches: Optional[re.Match[str]] = re.match(self.pattern, self.line)
+            # TODO: this is weird structure, attribute error is for if pattern or line is not defined?
```

Comment:
I don't like this double error thing

Reply:
I think it was probably for `.groupdict()` if `re.match` doesn't return anything. Your version is clearer. We can remove the try/except though now.